components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/IdentityProviderSecretsProcessorImpl.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.secret.mgt.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.model.ResolvedSecret;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretType;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS;
+
+public class IdentityProviderSecretsProcessorImpl implements IdentityProviderSecretsProcessor {
+
+    private final SecretManager secretManager;
+    private final SecretResolveManager secretResolveManager;
+
+    public IdentityProviderSecretsProcessorImpl() {
+
+        this.secretManager = new SecretManagerImpl();
+        this.secretResolveManager = new SecretResolveManagerImpl();
+    }
+
+    @Override
+    public IdentityProvider addOrUpdateIdpSecrets(IdentityProvider identityProvider) throws SecretManagementException {
+
+        for (FederatedAuthenticatorConfig fedAuthConfig : identityProvider.getFederatedAuthenticatorConfigs()) {
+            for (Property prop : fedAuthConfig.getProperties()) {
+                if (prop.isConfidential()) {
+                    String secretName =
+                            buildSecretName(identityProvider.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
```

Comment:
Let's try to avoid the code shifting to right.

Reply:
Resolved with [2b47e12](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/2b47e121b1c14f3ddce774f6b6a867c957fb5848)